pkg/diagnostics/grpc_tracing.go

```diff
@@ -67,6 +61,12 @@ func GRPCTraceUnaryServerInterceptor(appID string, spec config.TracingSpec) grpc
 			}
 		}
 
+		// Add trace context if absent
```

Comment:
This code will never hit because this is still incoming context. you need to check the response from handler call.

Reply:
you are almost there...we can discuss it.

Reply:
Oh that's right, changed it.